src/sentry/static/sentry/app/views/settings/components/dataPrivacyRules/rulesList.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import PropTypes from 'prop-types';
+
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import TextOverflow from 'app/components/textOverflow';
+import {IconDelete, IconEdit} from 'app/icons';
+import Button from 'app/components/button';
+
+import DataPrivacyRulesForm from './dataPrivacyRulesForm/dataPrivacyRulesForm';
+import {getRuleTypeLabel, getMethodTypeLabel} from './dataPrivacyRulesForm/utils';
+
+type Rule = React.ComponentProps<typeof DataPrivacyRulesForm>['rule'];
+type Props = {
+  rules: Array<Rule>;
+  onShowEditRuleModal?: (id: Rule['id']) => () => void;
+  onDeleteRule?: (id: Rule['id']) => () => void;
+};
+
+const RulesList = React.forwardRef<HTMLUListElement, Props>(function RulesList(
+  {rules, onShowEditRuleModal, onDeleteRule},
+  ref
+) {
+  return (
+    <List ref={ref}>
+      {rules.map(({id, method, type, source}) => {
+        const methodLabel = getMethodTypeLabel(method);
+        const typelabel = getRuleTypeLabel(type);
+        return (
+          <ListItem key={id}>
+            <TextOverflow>
+              {`[${methodLabel.label}] [${typelabel}] ${t('from')} [${source}]`}
+            </TextOverflow>
+            {onShowEditRuleModal && (
+              <Button
+                label={t('Edit Rule')}
+                size="small"
+                onClick={onShowEditRuleModal(id)}
+                icon={<IconEdit />}
+              />
+            )}
+            {onDeleteRule && (
+              <Button
+                label={t('Delete Rule')}
+                size="small"
+                onClick={onDeleteRule(id)}
+                icon={<IconDelete />}
+              />
+            )}
+          </ListItem>
+        );
+      })}
+    </List>
+  );
+});
+
+RulesList.propTypes = {
+  rules: PropTypes.array.isRequired,
+  onShowEditRuleModal: PropTypes.func,
+  onDeleteRule: PropTypes.func,
+};
```

Comment:
Do we have non-typescript components that use this?

Reply:
no. The reason I added the propTypes is because of lint warnings

Reply:
![image](https://user-images.githubusercontent.com/29228205/82411407-0475ce00-9a72-11ea-9610-d733b8d6769b.png)